tests/test_examples.py

```diff
@@ -236,118 +248,142 @@ def test_run_rbfe_legs(
     mol_b,
     seed,
 ):
-    with temporary_working_dir() as temp_dir:
-        with resources.as_file(resources.files("timemachine.datasets.fep_benchmark.hif2a")) as hif2a_dir:
-            config = dict(
-                mol_a=mol_a,
-                mol_b=mol_b,
-                sdf_path=hif2a_dir / "ligands.sdf",
-                pdb_path=hif2a_dir / "5tbm_prepared.pdb",
-                seed=seed,
-                legs=leg,
-                n_eq_steps=n_eq_steps,
-                n_frames=n_frames,
-                n_windows=n_windows,
-                # Use simple charges to avoid os-dependent charge differences
-                forcefield="smirnoff_1_1_0_sc.py",
+    # Can generate hashes from CI artifacts
+    endstate_hashes = {
+        "vacuum": (
+            "81fe2a16aa7eb89e6a05e1d4e162d24b57e4c0c3effe07f6ff548ca32618d8e6",
+            "9c05f938850f0a0f85643e76f6201e8519f9d1c5ceefdd380b677f394f0b35f1",
+        ),
+        "solvent": (
+            "6d8b39f723727d556b47e0907e89f1afe9843cae0b2e6107a745d1c29f9a3c8d",
+            "d6ecd7973f6f2c9d70f9652364e6b926f5551dcca6be0514bda3f0cd3524ccd0",
+        ),
+        "complex": (
+            "b3c18188bd8fe2475152b5c1c8b9d81799037f8c33685db5834943a140cc2988",
+            "931c428e50c47c2a5442f2fa027d9a4acaf140339db34e2972d8ec6d4f00b40b",
+        ),
+    }
+    with resources.as_file(resources.files("timemachine.datasets.fep_benchmark.hif2a")) as hif2a_dir:
+        config = dict(
+            mol_a=mol_a,
+            mol_b=mol_b,
+            sdf_path=hif2a_dir / "ligands.sdf",
+            pdb_path=hif2a_dir / "5tbm_prepared.pdb",
+            seed=seed,
+            legs=leg,
+            n_eq_steps=n_eq_steps,
+            n_frames=n_frames,
+            n_windows=n_windows,
+            # Use simple charges to avoid os-dependent charge differences
+            forcefield="smirnoff_1_1_0_sc.py",
+            output_dir=f"{ARTIFACT_DIR_NAME}/rbfe_{mol_a}_{mol_b}_{leg}_{seed}",
```

Comment:
Does each CI run get a fresh artifact directory? (not sure how this works on the gitlab side...)

If not, should we add an ID or timestamp to the path to avoid races with concurrent CI jobs?